apps/src/blockTooltips/DropletAutocompletePopupTooltipManager.js

```diff
@@ -0,0 +1,122 @@
+var DropletFunctionTooltipMarkup = require('./DropletFunctionTooltip.html.ejs');
+
+/**
+ * @fileoverview Displays tooltips for Droplet blocks
+ */
+
+/**
+ * Handles displaying tooltips on Droplet's ACE editor autocomplete dropdown
+ * @param {DropletTooltipManager} dropletTooltipManager
+ * @constructor
+ */
+var DropletAutocompletePopupTooltipManager = function (dropletTooltipManager) {
+  this.dropletTooltipManager = dropletTooltipManager;
+};
+
+var DEFAULT_TOOLTIP_CONFIG = {
+  interactive: true,
+  autoClose: false,
+  trigger: 'custom',
+  speed: 100,
+  maxWidth: 450,
+  position: 'left',
+  contentAsHTML: true,
+  theme: 'droplet-block-tooltipster',
+  offsetY: 2,
+  restoration: 'none',
+  updateAnimation: false
+};
+
+/**
+ * @param {Editor} dropletEditor
+ */
+DropletAutocompletePopupTooltipManager.prototype.registerHandlers = function (dropletEditor) {
+  if (!window.$) {
+    return; // TODO(bjordan): remove when $ available on dev server
+  }
+
+  var aceEditor = dropletEditor.aceEditor;
+
+  var registerOnEditorChanged = function (e) {
+    if (e.command.name !== 'insertstring') {
+      return;
+    }
+
+    var popupHasBeenShown = aceEditor.completer && aceEditor.completer.popup;
+    if (!popupHasBeenShown) {
+      return;
+    }
+
+    this.registerPopupHandlers_(aceEditor);
+    aceEditor.commands.off("afterExec", registerOnEditorChanged);
+  }.bind(this);
+
+  aceEditor.commands.on("afterExec", registerOnEditorChanged);
+};
+
+DropletAutocompletePopupTooltipManager.prototype.registerPopupHandlers_ = function (aceEditor) {
+  this.installAfterRenderHandler_(aceEditor);
+  this.installHideHandler_(aceEditor);
+};
+
+DropletAutocompletePopupTooltipManager.prototype.installAfterRenderHandler_ = function (aceEditor) {
+  aceEditor.completer.popup.setSelectOnHover(true);
+
+  aceEditor.completer.popup.renderer.on("afterRender", function () {
+    this.updateAutocompletePopupTooltip(aceEditor);
+  }.bind(this));
+};
+
+DropletAutocompletePopupTooltipManager.prototype.installHideHandler_ = function (aceEditor) {
+  aceEditor.completer.popup.on("hide", function () {
+    this.destroyAutocompleteTooltips_();
+  }.bind(this));
+};
```

Comment:
Feels a little like we might be using too many functions to do one set of things (as far as I can tell each of these functions is only being called in one place). I might refactor this in two ways:
(1) Move registerOnEditorChanged onto the prototype. In your calls to on/off you'll probably then need to do this.registerOnEditorChanged.bind(this)
(2) Move all this logic into registerOnEditorChanged. At the very least, I'd get rid of registerPopupHandlers_ and put it's two calls inside registerOnEditorChanged.